app/Services/AdminRoleServices/RoleService.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Services\AdminRoleServices;
+
+use App\Enums\GeneralEnums;
+use App\Exports\GeneralReportExport;
+use App\Helpers\GeneralHelper;
+use Carbon\Carbon;
+use Maatwebsite\Excel\Facades\Excel;
+use App\Models\Role;
+use Illuminate\Support\Str;
+use Spatie\Permission\Models\Permission;
+
+class RoleService
+{
+    public function overview($request)
+    {
+
+        $dateFilter = $request->date_filter;
+
+        if ($dateFilter === "1day") {
```

Comment:
Call GeneralHelper::dateFilter($request->date_filter) here. Anyone wey no dey the if block, add am